### PR TITLE
Migrator for renamed attributes

### PR DESCRIPTION
**Pull request by mikmakmuk**

### CODE REVIEW

File: bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java [hunk 1]
```diff
@@ -44,6 +45,23 @@ public static void executeStatements(List<String> queries, Connection connection
 		}
 	}
 	
+	public static int getIntResult(String query, int resultIndex, Connection connection) throws SpecmateException {
```
Comment (junkerm) [inline bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java]: There is no closing of the statement or the result set. Is this necessary?

File: bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java [hunk 2]
```diff
@@ -44,6 +45,23 @@ public static void executeStatements(List<String> queries, Connection connection
...
+		int res = 0;
+		try {
+			PreparedStatement st = SQLUtil.executeStatement(query, connection);
+			ResultSet result = st.getResultSet();
+			if (result.next()) {
+				res = result.getInt(resultIndex);
+			} else {
+				throw new SpecmateException("Could not get retrieve value.");
```
Comment (junkerm) [inline bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java]: Give a more specific exception text

File: bundles/specmate-migration/src/com/specmate/migration/h2/SQLMapper.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		this.packageName = packageName;
+		this.sourceVersion = sourceVersion;
+		this.targetVersion = targetVersion;
+	}
+
+	protected List<String> insertExternalReferences(String objectName, List<String> attributeNames) throws SpecmateException {
+		int id = SQLUtil.getIntResult("SELECT id FROM CDO_EXTERNAL_REFS ORDER BY id ASC LIMIT 1", 1, connection);
+		
+		List<String> queries = new ArrayList<>();
+		String baseUri = "http://specmate.com/" + targetVersion + "/" + packageName + "#//" + objectName;
```
Comment (junkerm) [inline bundles/specmate-migration/src/com/specmate/migration/h2/SQLMapper.java]: make "http://specmate.com" a constant

File: bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java (hunk 1 above)
Comment (mikmakmuk) [reply bundles/specmate-migration/src/com/specmate/migration/h2/SQLUtil.java]: Yes, good catch! Fixed.

File: bundles/specmate-migration/src/com/specmate/migration/h2/AttributeToSQLMapper.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	
+	public void migrateRenameAttribute(String objectName, String oldAttributeName, String newAttributeName) throws SpecmateException {
+		String failmsg = "Migration: Could not rename column " + oldAttributeName + " in table " + objectName + ".";
+		List<String> queries = new ArrayList<>();
+		queries.add("ALTER TABLE " + objectName + " ALTER COLUMN " + oldAttributeName + " RENAME TO " + newAttributeName);
+		queries.add(renameExternalReference(objectName, oldAttributeName, newAttributeName));
+		SQLUtil.executeStatements(queries, connection, failmsg);
+	}
+	
+	private void alterDB(String alterString, String objectName, String attributeName, Object defaultValue) throws SpecmateException {
```
Comment (junkerm) [inline bundles/specmate-migration/src/com/specmate/migration/h2/AttributeToSQLMapper.java]: Should you rename this method in something like "setDefaultAndExecute"?

Comment (mikmakmuk) [reply bundles/specmate-migration/src/com/specmate/migration/h2/AttributeToSQLMapper.java]: Done.

**Pull request merged**